org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/GradleDistribution.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2018 the original author or authors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.eclipse.buildship.core;
+
+import java.io.File;
+import java.net.URI;
+
+/**
+ * Represents a Gradle distribution.
+ *
+ * @author Donat Csikos
+ * @since 3.0
+ */
+public abstract class GradleDistribution {
+
+    /**
+     * The available Gradle distributions types.
+     */
+    public enum Type {
```

Comment:
❌ Should not be public, as previously discussed. What would be the purpose of making this public?